featuremanagement/_models/_feature_flag.py

```diff
@@ -15,60 +16,135 @@
 
 
 class FeatureConditions:
+    """
+    Represents the conditions for a feature flag
+    """
+
+    def __init__(self):
+        self._requirement_type = REQUIREMENT_TYPE_ANY
+        self._client_filters = []
+
     @classmethod
-    def convert_from_json(cls, json):
+    def convert_from_json(cls, json_value):
+        """
+        Convert a JSON object to FeatureConditions
+
+        :param json: JSON object
+        :type json: dict
+        :return: FeatureConditions
+        :rtype: FeatureConditions
+        """
         conditions = cls()
-        conditions._requirement_type = json.get(FEATURE_FILTER_REQUIREMENT_TYPE, REQUIREMENT_TYPE_ANY)
-        conditions._client_filters = json.get(FEATURE_FLAG_CLIENT_FILTERS, [])
+        if json_value is not None and not isinstance(json_value, Mapping):
+            raise AttributeError("Feature flag conditions must be a dictionary")
+        conditions._requirement_type = json_value.get(FEATURE_FILTER_REQUIREMENT_TYPE, REQUIREMENT_TYPE_ANY)
+        conditions._client_filters = json_value.get(FEATURE_FLAG_CLIENT_FILTERS, [])
         return conditions
 
     @property
     def requirement_type(self):
+        """
+        Get the requirement type for the feature flag
+
+        :return: Requirement type
+        :rtype: str
+        """
         return self._requirement_type
 
     @property
     def client_filters(self):
+        """
+        Get the client filters for the feature flag
+
+        :return: Client filters
+        :rtype: list[dict]
+        """
         return self._client_filters
 
     def _validate(self, feature_flag_id):
         if self._requirement_type not in [REQUIREMENT_TYPE_ALL, REQUIREMENT_TYPE_ANY]:
-            raise ValueError("Feature flag {} has invalid requirement type.".format(feature_flag_id))
+            raise ValueError(f"Feature flag {feature_flag_id} has invalid requirement type.")
         for feature_filter in self._client_filters:
             if feature_filter.get(FEATURE_FILTER_NAME) is None:
-                raise ValueError("Feature flag {} is missing filter name.".format(feature_flag_id))
+                raise ValueError(f"Feature flag {feature_flag_id} is missing filter name.")
+
 
 class FeatureFlag:
+    """
+    Represents a feature flag
+    """
+
+    def __init__(self):
```

Comment:
I don't feel too strongly either way, but can we tell the linter to not require these init functions if we don't really use them?

Reply:
We can, but either we need ignores on every method that does have the first use of a class variable, or we would have to set it as a global rule, see the other change I just made.

I actually don't mind them as they make it clear what default values are, though it will be more clear when we add Variant support where this gets a lot more complex.